### PR TITLE
fix: convert from FastAPI to Flask

### LINKED ISSUE #40: use Flask instead of FastAPI
My first attempt to build this tutorial used [Flask](https://flask.palletsprojects.com/) as a server rather than [FastAPI](https://fastapi.tiangolo.com/). While working on the authentication lesson, I'm wondering if I should have stayed with Flask. For example, managing redirect-on-login in FastAPI is more complex than doing it in Flask, and while FastAPI is better for serving JSON for REST APIs, this tutorial is mostly serving HTML.

### PR DESCRIPTION
**Pull request by gvwilson**
closes #40

### CODE REVIEW

**Comment by gvwilson** [comment]
@juananpe this one is ready to review - please ignore all the changes in `./docs` (I re-rendered).

File: docs/data/assays.json
Comment (juananpe) [inline docs/data/assays.json:1]: This change is not related to FastAPI->Flask

File: docs/12_perm/explore_permissions.py
Comment (juananpe) [inline docs/12_perm/explore_permissions.py:1]: This change is not related to FastAPI->Flask

File: docs/11_migrate/migrate.py
Comment (juananpe) [inline docs/11_migrate/migrate.py:1]: This change is not related to FastAPI->Flask

**Review by juananpe — COMMENTED** [review]
Good job with the conversion process! A lot of changes but it looks good (just some minor things commented in the review)

File: docs/09_browser/server.py [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
     """Build application and configure routes."""
-    app = FastAPI()
-    app.add_middleware(
-        CORSMiddleware,
-        allow_origins=["*"],
-        allow_credentials=True,
-        allow_methods=["*"],
-        allow_headers=["*"],
-    )
+    app = Flask("err")
```
[the remark this answers is not in the document]
Comment (gvwilson) [reply docs/09_browser/server.py:10]: we would then have to change all of the subsequent examples as well - I think it's simpler at this stage to leave flask on port 5000 (its default) and serve rendered web pages on port 8000. I would like to add a later lesson (after authentication and internationalization) to talk about coordinating multiple services, setting ports, etc., and I think that would be a better place to do this.
